src/scratch/ScratchRuntime.as

```diff
@@ -77,8 +85,56 @@ public class ScratchRuntime {
 			saveAfterInstall = false;
 			return;
 		}
-
-		if (recording) saveFrame(); // Recording a YouTube video?  Old / Unused currently.
+		if (ready==0) {
+			var tR:Number = getTimer()*.001-secs;
+			while (t>sounds.length/framerate+1/framerate) {
+				saveSound();
+			}
+			var count:int = 3;
+			if (tR>=3.75){
+				ready = 1;
+				count = 1;
+				sounds = [];
+				frames=[];
+				Scratch.app.log(LogLevel.TRACK, "Project video started");
+			}
+			else if (tR>=2.5){
+				count=1
+			}
+			else if (tR>=1.25 && mReady) {
+				count=2;
+			}
+			else if (tR>=1.25) {
+				secs+=tR;
+			}
+			else {
+				app.refreshStagePart();
+			}
+		}
+		if (recording) { // Recording a YouTube video?
```

Comment:
I've made some comments in the if (recording) section to clarify what's going on - let me know if they're unclear.